src/main/java/in/projecteka/consentmanager/user/SignUpRequestValidator.java

```diff
@@ -106,11 +97,10 @@ private SignUpRequestValidator() {
                                           seq.distinct().sorted())));
     }
 
-    private static Validation<String, LocalDate> validateAge(LocalDate dateOfBirth) {
-        return dateOfBirth == null ||
-                       dateOfBirth.isBefore(TOMORROW)
-               ? Validation.valid(dateOfBirth)
-               : Validation.invalid("Date of birth can't be in future");
+    private static Validation<String, Integer> validateYearOfBirth(int year) {
+        return ((year <= (TODAY.getYear())) && (year >= TODAY.getYear() - 120))
+                ? Validation.valid(year)
+                : Validation.invalid("Year of birth can't be in future or older than 120 years");
     }
```

Comment:
YearOfBirth is optional right?

Reply:
yes

Reply:
It would be better if you make the validation obvious. Use Integer null or do the limit validation? Make sense?